src/haxelib/api/Vcs.hx

```diff
@@ -420,4 +437,9 @@ class Mercurial extends Vcs {
 		if (run(vcsArgs, debugLog).code != 0)
 			throw VcsError.CantCloneRepo(this, url/*, ret.out*/);
 	}
+
+	public function getReproducibleVersion(libPath: String): VcsData {
+		// TODO
+		throw new haxe.exceptions.NotImplementedException("Getting revision hash isn't supported for Mercurial");
```

Comment:
Ok so I was able to get an implementation:
```haxe
		final oldCwd = Sys.getCwd();
		Sys.setCwd(libPath);

		final ret: VcsData = {
			url: run(["paths", "default"], true).out.trim(),
			ref: run(["identify", "-i", "--debug"], true).out.trim(),
		};

		Sys.setCwd(oldCwd);
		return ret;
```
**But** that doesn't work for installation, since `-L foo:hg:...#rev` will try to install `rev` as being a branch exclusively.

I had to update `clone` too:
```diff
        public function clone(libPath:String, url:String, ?branch:String, ?version:String, ?debugLog:(msg:String)->Void):Void {
                final vcsArgs = ["clone", url, libPath];
 
-               if (branch != null) {
+               if (branch != null && version != null) {
                        vcsArgs.push("--branch");
                        vcsArgs.push(branch);
-               }
 
-               if (version != null) {
                        vcsArgs.push("--rev");
                        vcsArgs.push(version);
+               } else if (branch != null) {
+                       vcsArgs.push("--updaterev");
+                       vcsArgs.push(branch);
+               } else if (version != null) {
+                       vcsArgs.push("--updaterev");
+                       vcsArgs.push(version);
                }
 
                if (run(vcsArgs, debugLog).code != 0)
                        throw VcsError.CantCloneRepo(this, url/*, ret.out*/);
        }
```

But I'd like to add proper tests for that. Only tested with `https://www.mercurial-scm.org/repo/hello` locally with:
```sh
haxelib hg test-tag https://repo.mercurial-scm.org/hello tip
haxelib hg test-branch https://repo.mercurial-scm.org/hello default
haxelib hg test-branch-tag https://repo.mercurial-scm.org/hello default "" tip
haxelib hg test-rev https://repo.mercurial-scm.org/hello 82e55d328c8ca4ee16520036c0aaace03a5beb65
```

Which all worked